digits/frameworks/framework.py

```diff
@@ -26,6 +26,18 @@ def can_shuffle_data(self):
         """
         return self.CAN_SHUFFLE_DATA
 
+    def supports_python_layers_file(self):
+        """
+        return whether framework can shuffle input data during training
+        """
+        return self.SUPPORTS_PYTHON_LAYERS_FILE
+
+    def supports_timeline_traces(self):
```

Comment:
Ah yeah, timeline traces. Those were pretty fancy. Did you ever test this? You don't need to.